x-pack/plugins/security_solution/scripts/endpoint/agent_downloader_cli/agent_downloader.ts

```diff
@@ -11,21 +11,63 @@ import type { ToolingLog } from '@kbn/tooling-log';
 import { getAgentDownloadUrl, getAgentFileName } from '../common/fleet_services';
 import { downloadAndStoreAgent } from '../common/agent_downloads_service';
 
+// Decrement the patch version by 1
+const decrementPatchVersion = (version: string): string => {
+  const [major, minor, patch] = version.split('.').map(Number);
+  return `${major}.${minor}.${patch - 1}`;
+};
+
+// Generate a list of versions to attempt downloading, including a fallback to the previous patch (GA)
+const getVersionsToDownload = (version: string): string[] => {
+  const patch = parseInt(version.split('.')[2], 10);
```

Comment:
Please use `semver.patch(version)` (may need to ensure that `version` does not have `-SNAPSHOT` in it)

Reply:
Good hint, I'll use semver here, and drop the pre-release tag